### DISCUSSION TITLE
How to create and write a PATH STRNAME in GDS?

### DISCUSSION THREAD

**Discussion by ghost** [gdsfactory/gdsfactory#3414] [Q&A] [answered]
If I `gf.import_gds()` a GDS file with a PATH structure made in Klayout, I can view it, and then `gf.write()` and the PATH structure is preserved, so I assume that PATH structures are supported?

Per a previous question (https://github.com/gdsfactory/gdsfactory/discussions/2171). As [Chuplin](https://github.com/Chuplin) [mentioned](https://github.com/gdsfactory/gdsfactory/discussions/2171#discussioncomment-9283361) there is EBL machine code that recognize the PATH structure. I too would like to generate a path! Have I overlooked some documentation that might provide an answer?

 

Path, but not really a PATH:
```python
import gdsfactory as gf
gf.CONF.max_cellname_length = 99

@gf.cell
def point(length:float=1) -> gf.Component:      
        a = gf.path.straight(length=length)
        b = gf.path.extrude(a, layer=(3,0), width=1)
        return b

@gf.cell
def point_array(component:gf.Component,
                cols:int,
                rows:int,
                column_pitch:float,
                row_pitch:float,
                ) -> gf.Component:
        
        c = gf.Component()
        c.add_ref(component=component,
                columns = cols,
                rows = rows,
                column_pitch=column_pitch,
                row_pitch=row_pitch,
                )
        return c

p = point(1)
pa = point_array(p,10,3,5,5)
pa.write_gds(gdspath = 'gfpath.gds', with_metadata=False)
'''Output of the path is a BOUNDARY aka polygon'''
```

My attempt (not really an attempt, but a place to start from):
```python
import gdsfactory as gf

c1 = gf.kdb.Path(pts = [[0,0],[600,0],[600,400]], width=2)
'''Create a layout somehow...'''
c3= gf.Component()
ref1 = c2.add_ref(c1)
```

As mentioned, since I can view, and re-export PATH structures, I assume I can create them, but I do not understand how I would go about generating a component from a `gf.kdb.Path()`.

**Comment by sebastian-goeldi** [comment, marked as answer]
A path is just that, a path, not a component/cell which you can instantiate. Since you want to use the klayout objects, you need to treat it like one, and use klayput API to add it to a component and not gdsfactory API as it doesn't provide means to add pathes/boxes to my knowledge.

```python
import gdsfactory as gf


path = gf.kdb.Path(pts = [[0,0],[600,0],[600,400]], width=2) # this is valid but undocumented behavior, otherwise use a list of kdb.Point for properly documented constructors.

c = gf.Component()
c.shapes(LAYER.WG).insert(path)
```

Also, as a side note:

Polygon/Box/Path/Edge/EdgePair are all in dbu (meaning nm in the default settings). If you want to define in ums, use their D counterpart, e.g. `DPath`
